google/cloud/spanner/client.h

```diff
@@ -491,8 +491,9 @@ class Client {
    * @par Example
    * @snippet samples.cc execute-batch-dml
    */
-  StatusOr<BatchDmlResult> ExecuteBatchDml(
-      Transaction transaction, std::vector<SqlStatement> statements);
+  StatusOr<BatchDmlResult> ExecuteBatchDml(Transaction transaction,
+                                           std::vector<SqlStatement> statements,
+                                           QueryOptions const& opts = {});
```

Comment:
Do other `QueryOptions` settings, like the optimizer version, make sense for these DML methods?

Reply:
For `ExecutePartionedDml()`, yes, because it is implemented using `ExecuteSqlRequest`, which at least has space for `optimizer_version`.  That said, I don't know whether `optimizer_version` actually applies in Partitioned DML mode.

But for `ExecuteBatchDml()`, no ... it is implemented using `ExecuteBatchDmlRequest`, which does not have space for `optimizer_version`.

The alternative seemed to be to introduce a new `DmlOptions`, with just `priority`, for at least `ExecuteBatchDml()` (and for `ExecutePartitionedDml()` if `optimizer_version` doesn't apply there).  But that seemed counter to our new "options bucket" philosophy.   Still, I'm happy either way (or some other way).

Reply:
I wouldn't want to introduce a _new_ options type, but perhaps using `google::cloud::Options` would make sense.